src/cryptenroll/cryptenroll.c

```diff
@@ -636,7 +676,15 @@ static int prepare_luks(
         assert(ret_cd);
         assert(!ret_volume_key == !ret_volume_key_size);
 
-        r = crypt_init(&cd, arg_node);
+        if (!arg_node) {
+                r = resolve_auto_node(&auto_node);
+                if (r == -EMEDIUMTYPE)
+                        return log_error_errno(r, "Block device backing /var isn't a LUKS2 device. Please specify a device node manually");
+                if (r < 0)
+                        return log_error_errno(r, "Failed to automatically resolve device node: %m");
+        }
```

Comment:
i'd just do:

```c
arg_node = TAKE_PTR(auto_node);
```

here, and continue on with arg_node below

Reply:
Can't do, since this conflicts with the future change where I split parts of this code into a separate function to be used from varlink (where we don't set `arg_node`)